integration/yarn/src/main/java/alluxio/yarn/ContainerAllocator.java

```diff
@@ -153,7 +153,7 @@ private void requestContainers(int numContainersToRequest) throws Exception {
       priority = Priority.newInstance(101);
     }
 
-    if (hosts.length * mMaxContainersPerHost < numContainersToRequest) {
+    if (hosts.length * mMaxContainersPerHost < mTargetNumContainers - mAllocatedContainers.size()) {
```

Comment:
We can completely remove the `numContainersToRequest` argument and use `mTargetNumContainers - mAllocatedContainers.size()` instead